contracts/Agreement.sol

```diff
@@ -21,23 +61,52 @@ contract Agreement {
         address indexed accepter
     );
 
+    event LogAgreementAmended(
+        address platformOperatorRepresentative,
+        string agreementUri
+    );
+
     ////////////////////////
     // Modifiers
     ////////////////////////
 
+    /// @notice logs that agreement was accepted by platform user
+    /// @dev intended to be added to functions that if used make 'accepter' origin to enter legally binding agreement
     modifier acceptAgreement(address accepter) {
-        LogAgreementAccepted(accepter);
+        require(_amendments.length > 0);
+        if(!_signatories[accepter]) {
+            _signatories[accepter] = true;
+            LogAgreementAccepted(accepter);
+        }
```

Comment:
We can safely move the `require` inside the conditional, because no-one can be accepted before the first amendment is posted.
This doesn't look as good, but it saves an expensive storage lookup in the common case where user has already signed. I think it's worth it.